### PR TITLE
feat: lint & test pass

### PR DESCRIPTION
**Pull request by dejanzele**
## Pull request description 

Propagate context object
Fix resource leak in defer in a loop
Propagate error for EndExecution

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: internal/app/api/v1/executions.go [hunk 1]
```diff
@@ -365,7 +364,7 @@ func (s TestkubeAPI) ListArtifactsHandler() fiber.Handler {
 // streamLogsFromResult writes logs from the output of executionResult to the writer
 func (s *TestkubeAPI) streamLogsFromResult(executionResult *testkube.ExecutionResult, w *bufio.Writer) error {
 	enc := json.NewEncoder(w)
-	fmt.Fprintf(w, "data: ")
+	_, _ = fmt.Fprintf(w, "data: ")
```
Comment (vsukhin) [inline internal/app/api/v1/executions.go:367]: I would log errors in any case

File: pkg/executor/common.go [hunk 2]
```diff
@@ -181,14 +184,14 @@ func GetJobPods(podsClient tcorev1.PodInterface, jobName string, retryNr, retryC
...
+		return GetJobPods(ctx, podsClient, jobName, retryNr+1, retryCount)
 	}
 	return pods, nil
 }
 
 // GetPodLogs returns pod logs bytes
-func GetPodLogs(c kubernetes.Interface, namespace string, pod corev1.Pod, logLinesCount ...int64) (logs []byte, err error) {
-	count := int64(100)
+func GetPodLogs(ctx context.Context, c kubernetes.Interface, namespace string, pod corev1.Pod, logLinesCount ...int64) (logs []byte, err error) {
+	var count int64 = 100
```
Comment (vsukhin) [inline pkg/executor/common.go:194]: Is 100  a const?

**Review by vsukhin — APPROVED** [review]
looks good, two minor comments

File: pkg/executor/common.go (hunk 2 above)
Comment (dejanzele) [reply pkg/executor/common.go:194]: I don't know, this was from before, I just changed the semantics. Do you maybe know?

Comment (vsukhin) [reply pkg/executor/common.go:194]: No, I didn't see it before. Put 100 to const at the beginning of the file

**Pull request merged**